exercise/views.py

```diff
@@ -42,10 +45,27 @@ def get_permissions(self):
     @method_decorator(cache_page(TWO_HOURS))
     def init(self, request, *args, **kwargs):
         exercise = self.get_object()
-        participant = Participant(exercise=exercise)
-        participant.save()
+        participant = create_participant(exercise, None)
+        resp = {
+            "participant": str(participant.id),
+            "exercise": ExerciseSerializer(exercise).data,
+        }
+        return Response(data=resp)
+
+    @action(
+        methods=["GET"],
+        detail=False,
+        permission_classes=[],
+        url_path="(?P<exercise_id>.+)/init/(?P<trial_id>.+)",
+    )
+    @method_decorator(cache_page(TWO_HOURS))
+    def trial(self, request, *args, **kwargs):
+        exercise = get_object_or_404(Exercise, id=kwargs["exercise_id"])
+        participant = create_participant(exercise, kwargs["trial_id"])
+
         resp = {
             "participant": str(participant.id),
+            "trial": str(participant.trial),
```

Comment:
this will return the name of the trial rather than the number, is that intentional?

Reply:
Yep